daemon/container.go

```diff
@@ -1286,7 +1286,7 @@ func (container *Container) verifyDaemonSettings() {
 		logrus.Warnf("Your kernel does not support memory limit capabilities. Limitation discarded.")
 		container.Config.Memory = 0
 	}
-	if container.Config.Memory > 0 && !container.daemon.sysInfo.SwapLimit {
+	if container.Config.MemorySwap > 0 && !container.daemon.sysInfo.SwapLimit {
```

Comment:
Config.Memory > 0 means we'll use memorySwap as well, and Config.MemorySwap might be 0.
I think we can fix it by:
if container.Config.Memory > 0 && container.Config.MemorySwap != -1 && !container.daemon.sysInfo.SwapLimit {